kedro/framework/cli/catalog.py

```diff
@@ -27,11 +28,6 @@ def _create_session(package_name: str, **kwargs: Any) -> KedroSession:
     return KedroSession.create(**kwargs)
 
 
-def is_parameter(dataset_name: str) -> bool:
```

Comment:
Good catch, I returned `is_parameter` to `kedro/framework/cli/catalog.py`. Since we do not want to import from high-level to low-level components I added `_is_parameter` to `kedro/io/core.py` and `TODOs` to replace it when breaking change.